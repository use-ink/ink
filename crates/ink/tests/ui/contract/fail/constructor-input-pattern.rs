#![allow(unexpected_cfgs)]

#[ink::contract]
mod contract {
    #[ink(storage)]
    pub struct Contract {}

    impl Contract {
        #[ink(constructor)]
        pub fn constructor((_a, _b): (i32, i32)) -> Self {
            Self {}
        }

        #[ink(message)]
        pub fn message(&self) {}
    }
}

fn main() {}
