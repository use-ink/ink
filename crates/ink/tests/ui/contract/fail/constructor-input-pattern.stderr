error: ink! constructor arguments must have an identifier
  --> tests/ui/contract/fail/constructor-input-pattern.rs:10:28
   |
10 |         pub fn constructor((_a, _b): (i32, i32)) -> Self {
   |                            ^^^^^^^^^^^^^^^^^^^^
